components/brave_wallet/renderer/js_ethereum_provider.cc

```diff
@@ -164,12 +166,6 @@ void JSEthereumProvider::Install(bool allow_overwrite_window_ethereum_provider,
 
   // Check window.ethereum existence.
   v8::Local<v8::Object> global = context->Global();
-  v8::Local<v8::Value> ethereum_value =
-      global->Get(context, gin::StringToV8(isolate, kEthereum))
```

Comment:
we need to check `kBraveEthereum` here

Reply:
`install_brave_ethereum_provider ` should also be checked here before we actually create the provider object.

Reply:
as I mentioned in Slack, with check before calling `Install()`, we don't need to pass `install_brave_ethereum_provider` anymore which also simplifies logic in Install()

Reply:
Also the reason that we check very early in Install for existence is because `DidClearWindowObject` will be called multiple times under some scenario and we don't want to create provider object when it is already created